Add rolling desktop-full

chapulina opened the pull request:
Desktop-full is finally on ROS 2, see

* https://github.com/ros-infrastructure/rep/pull/334
* https://github.com/ros2/variants/pull/34

`variants` hasn't been bloomed with the new package yet, but I wanted to make sure I opened this PR before I forgot. The exact hardcoded version will likely change.

mikaelarguedas commented:
Great thanks!

Why not creating docker images for `perception` and `simulation` like it was the case in ROS 1 before ?
As this change has been made on REP-2001 that cover all ROS2 distros, this means that these new stacks are available in all ROS 2 distros ? (from REP-2001 it seems that desktop_full is available only un Humble but perception and simulation are available for all distris, is that right ?)

---
Most likely this will be superseeded by a PR adding the new images in a generic way to rolling and all future ros distros.

tfoote commented:
It would be good to add the other variants as well as base images. We can probably map docker image names to the REP variants.

chapulina commented:
> The exact hardcoded version will likely change.

Updated to 0.10.0-1, see 

* https://github.com/ros/rosdistro/pull/33173

> Why not creating docker images for perception and simulation like it was the case in ROS 1 before ?

Good point, added in [f26b990](https://github.com/osrf/docker_images/pull/619/commits/f26b990a491c91c212ed9db9ae1cc66240b38d69)

>  from REP-2001 it seems that desktop_full is available only un Humble but perception and simulation are available for all distris, is that right ?

Ouch, I can see the confusion. See https://github.com/ros-infrastructure/rep/pull/353. The answer is that they're all only available from Humble.

> Most likely this will be superseeded by a PR adding the new images in a generic way to rolling and all future ros distros.

Feel free to close this if https://github.com/osrf/docker_images/pull/620 gets merged before, thanks!

mikaelarguedas commented:
Superseeded by https://github.com/osrf/docker_images/pull/620
